### ISSUE TITLE
Focus loss and options menu stuck open when using preferences modal

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#28089]
**Describe the bug**
A focus loss occurs and the editor options menu gets stuck open when using the preferences modal. The issue with the menu being stuck open is caused by the focus loss.

**To reproduce**
Using the keyboard:
1. Open the Options menu
2. Navigate to Preferences and open that
3. Press escape to close Preferences
4. Observe that focus didn't return to the Options menu
5. Press escape, observe the options menu doesn't close
6. Using the mouse, click outside the option menu, observe the options menu

**Expected behavior**
Focus should return, and the options menu should be consistently closeable.

**Editor version (please complete the following information):**
- WordPress version: 5.6
- If the Gutenberg plugin is installed, which version is it? 9.7 (but on the `master` branch at b5e0e421af1890577b3b45cff42d49610e1d82d9)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]: Mac OS
 - Browser [e.g. chrome, safari]: Brave

**Comment by brentswisher** [comment]
I started looking into this but was unable to find a solution. I thought it worth noting that this is happening for any modal that is opened from the options menu. (Block Manager, Keyboard Shortcuts, Welcome Guide) 

Reading the docs I thought that [withFocusReturn](https://developer.wordpress.org/block-editor/components/with-focus-return/) would be able to be implemented here, but I am not able to get it working. For some reason even when wrapping the component in it, onFocusReturn() doesn't appear to be firing. I think it may possibly be due to the `PreferencesModal` component not actually unmounting, but being hidden with an isActive flag? I'll try to look into it further, but if anyone with more knowledge of withFocusReturn had any advice it would be appreciated.

**Comment by talldan** [comment]
@brentswisher There's now a `useFocusReturn` react hook that should be in use on the modal already, that's probably where things are not working as expected.

**Comment by brentswisher** [comment]
@talldan Thank you! There is a bug in useFocusReturn that is causing the issue, I should be able to get a PR with a fix soon.

**Comment by talldan** [comment]
Great, thanks for fixing it @brentswisher!
